ZIMP_game/models/game.py

```diff
@@ -1,27 +1,189 @@
+            return False
+
+    def update_current_tile_position(self, direction):
+        """Updates current tile position based on exit direction"""
+        match direction:
+            case "north":
+                self.current_tile = [self.current_tile[0] - 1, self.current_tile[1]]
+            case "east":
+                self.current_tile = [self.current_tile[0], self.current_tile[1] + 1]
+            case "south":
+                self.current_tile = [self.current_tile[0] + 1, self.current_tile[1]]
+            case "west":
+                self.current_tile = [self.current_tile[0], self.current_tile[1] - 1]
+            case _:
+                raise ValueError("Incorrect direction name!")
+        # Add to game map
+        self.game_map[self.current_tile[0],
+                      self.current_tile[1]] = self.current_tile_object
+
+    def revisit_tile(self, direction):
+        """Revisits a tile"""
+        match direction:
+            case "north":
+                self.current_tile = [self.current_tile[0] - 1, self.current_tile[1]]
+            case "east":
+                self.current_tile = [self.current_tile[0], self.current_tile[1] + 1]
+            case "south":
+                self.current_tile = [self.current_tile[0] + 1, self.current_tile[1]]
+            case "west":
+                self.current_tile = [self.current_tile[0], self.current_tile[1] - 1]
+            case _:
+                raise ValueError("Incorrect direction name!")
+
+        self.current_tile_object = self.game_map[self.current_tile[0],
+                                                 self.current_tile[1]]
+
+    def is_all_exits_explored(self):
+        """Returns true if all available exits are explored"""
+        # ADJACENT_TILE = 1
+        # Y_COORD = self.current_tile[0]
+        # X_COORD = self.current_tile[1]
+        empty_tile = 0
+        all_exits_explored = False
+
+        # if (self.game_map[Y_COORD + ADJACENT_TILE][X_COORD] != NO_TILES
+        #     & self.game_map[Y_COORD - ADJACENT_TILE][X_COORD] != NO_TILES
+        #     & self.game_map[Y_COORD][X_COORD + ADJACENT_TILE] != NO_TILES
+        #     & self.game_map[Y_COORD][X_COORD - ADJACENT_TILE] != NO_TILES):
+        #     all_exits_explored = True
+
+        for direction, is_exit in self.current_tile_object.exits:
+            if is_exit:
+                if self.generate_temp_location(direction) == empty_tile:
+                    return all_exits_explored
+        all_exits_explored = True
+
+        return all_exits_explored
+
+    def break_room(self):
+        """After the player battles the zombies and still have health left,
+        the player chooses a new direction to exit"""
+        NO_HEALTH = 0
+        self.battle()
+        if Player.health > NO_HEALTH:
+            valid_direction = False
+            direction = ""
+            while not valid_direction:  # And not entry direction!
+                direction = input("Please enter new direction.")

```

Comment:
If we use MVC structure any input from the user would be done in a View not model. Remove the input method and replace it with a class string variable that can be changed.